tests/test_time_slicing.py

```diff
@@ -125,22 +190,138 @@ def test_time_slicing_analysis(
     # target_states: (ar_steps, N_grid, d_features)
-    # forcing: (ar_steps, N_grid, d_windowed_forcing)
+    # forcing: (ar_steps, N_grid, d_windowed_forcing * 2)
     # target_times: (ar_steps,)
-    assert init_states.shape == (2, 1, 1)
-    assert init_states[:, 0, 0].tolist() == expected_init_states
 
-    assert target_states.shape == (3, 1, 1)
-    assert target_states[:, 0, 0].tolist() == expected_target_states
+    # Adjust assertions to use computed expected values
+    assert init_states.shape == (INIT_STEPS, 1, 1)
+    np.testing.assert_array_equal(init_states[:, 0, 0], expected_init_states)
+
+    assert target_states.shape == (ar_steps, 1, 1)
+    np.testing.assert_array_equal(
+        target_states[:, 0, 0], expected_target_states
+    )
 
     assert forcing.shape == (
-        3,
+        ar_steps,
         1,
-        1 + num_past_forcing_steps + num_future_forcing_steps,
+        total_forcing_window
+        * 2,  # Each windowed feature includes temporal embedding
+    )
+
+    # Extract the forcing values from the tensor (excluding temporal embeddings)
+    forcing_values = forcing[:, 0, :total_forcing_window]
+
+    # Compare with expected forcing values
+    for i in range(ar_steps):
+        np.testing.assert_array_equal(
+            forcing_values[i], expected_forcing_values[i]
+        )
+
+
+@pytest.mark.parametrize(
+    "ar_steps,num_past_forcing_steps,num_future_forcing_steps",
+    SCENARIOS,
+)
+def test_time_slicing_forecast(
+    ar_steps, num_past_forcing_steps, num_future_forcing_steps
+):
+    # Constants for forecast data
+    ANALYSIS_TIMES = np.datetime64("2020-01-01") + np.arange(
+        len(STATE_VALUES_FORECAST)
+    )
+    ELAPSED_FORECAST_DURATION = np.timedelta64(0, "D") + np.arange(
+        len(FORCING_VALUES_FORECAST[0])
+    )
+    # Create a dummy datastore with forecast data
+    time_values = (ANALYSIS_TIMES, ELAPSED_FORECAST_DURATION)
+    datastore = SinglePointDummyDatastore(
+        state_data=STATE_VALUES_FORECAST,
+        forcing_data=FORCING_VALUES_FORECAST,
+        time_values=time_values,
+        is_forecast=True,
     )
-    np.testing.assert_equal(forcing[:, 0, :], np.array(expected_forcing_values))
+
+    dataset = WeatherDataset(
+        datastore=datastore,
+        datastore_boundary=None,
+        split="train",
+        ar_steps=ar_steps,
+        num_past_forcing_steps=num_past_forcing_steps,
+        num_future_forcing_steps=num_future_forcing_steps,
+        standardize=False,
+    )
+
+    # Test the dataset length
+    assert len(dataset) == len(ANALYSIS_TIMES)
+
+    sample = dataset[0]
+
+    init_states, target_states, forcing, _, _ = [
+        tensor.numpy() for tensor in sample
+    ]
+
+    # Compute expected initial states and target states based on ar_steps
+    offset = max(0, num_past_forcing_steps - INIT_STEPS)
+    init_idx = INIT_STEPS + offset
+    expected_init_states = STATE_VALUES_FORECAST[0][offset:init_idx]
```

Comment:
These tests are so great. I was just thinking it might be nice to make it explicit here that this `0`-index is due to the fact we only have implemented sampling from the first ensemble member. That is the reason why? Then it will be clearer how to update the tests once we add sampling more ensemble members

Reply:
No these are indexing `analysis_time` as can be seen here: 
https://github.com/sadamov/neural-lam/blob/98c54d9856f4dead1ed8fd8951a69795931cb5e6/tests/test_time_slicing.py#L111

I added some comments to the tests to make this indexing clear. 4a278fd9dea3fd8ce7bce91e88cb6259fa42cc6a Ensemble functionality must be added at a later date.